src/main/webapp/src/components/autocompleteTextField.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { withStyles } from "@material-ui/styles";
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Typography from '@material-ui/core/Typography';
+import throttle from 'lodash/throttle';
+import axios from "axios";
+
+/**
+ * Change TextField outline border default colors.
+ */
+const StyledTextField = withStyles({
+  root: {
+    "& label.Mui-focused": {
+      color: "black",
+    },
+    "& .MuiInput-underline:after": {
+      borderBottomColor: "#733F94",
+    },
+    "& .MuiOutlinedInput-root": {
+      "&.Mui-focused fieldset": {
+        borderColor: "#733F94",
+      },
+    },
+  },
+})(TextField);
+
+export default function AutocompleteTextField(props) {
+  const [value, setValue] = React.useState(props.value);
+  const [options, setOptions] = React.useState([]);
+
+  const fetch = React.useMemo(
+    () =>
+      throttle((request, callback) => {
+        axios
+          .post(`/autocomplete?type=${props.type}&searchString=${request}`)
```

Comment:
In POST requests, we don't pass the parameters in the URL. For `axios`, I'm pretty sure the syntax would be `post('/autocomplete', { type: props.type, searchString: request })`, but maybe you need to adjust it slightly.